src/care/DebugPlugin.cpp

```diff
@@ -0,0 +1,134 @@
+#include "care/DebugPlugin.h"
+#include "chai/ExecutionSpaces.hpp"
+#include "care/CHAICallback.h"
+#include "care/PluginData.h"
+
+// Std library headers
+#if defined(CARE_DEBUG) && !defined(_WIN32)
+#include <execinfo.h>
+#endif // defined(CARE_DEBUG) && !defined(_WIN32)
+
+#include <unordered_set>
+
+namespace care{
+   void DebugPlugin::registerPlugin() {
+      static RAJA::util::PluginRegistry::add<care::DebugPlugin> L ("Debug plugin", "CARE plugin for debugging");
+   }
+
+   void DebugPlugin::preLaunch(const RAJA::util::PluginContext& p) {
+#if !defined(CHAI_DISABLE_RM)
+      // Prepare to record CHAI data
+      if (CHAICallback::isActive()) {
+         PluginData::clearActivePointers();
+
+#if defined(CARE_GPUCC) && defined(CARE_DEBUG)
+         GPUWatchpoint::setOrCheckWatchpoint<int>();
```

Comment:
@robinson96, can we remove the GPUWatchpoint stuff? Since we are not passing any address to it, I don't see how it can do anything useful.

Reply:
Oh, I see - you'd have to manually set a watchpoint before the loop. But I looked it up and cuda-gdb at least has the ability to set a watchpoint on both host and GPU memory. So I don't think we need this watchpoint code.

Reply:
Oh, wait. cuda-gdb only supports watchpoints on host memory. Sad day.

Reply:
The question still remains, though, is this used enough to keep it around? I've never used it myself.

Reply:
Looks like rocgdb supports watchpoints on device memory: https://docs.amd.com/projects/ROCdbgapi/en/docs-5.0.0/doxygen/docBin/html/group__watchpoint__group.html

Reply:
I guess we can leave the support in for now and it would be easy to remove later if we decide to get rid of it.